### DISCUSSION TITLE
The signed distance from a given point to a 2D slice of a given 3D mesh (for example, obj file)

### DISCUSSION THREAD

**Discussion by Pupillen** [CGAL/cgal#7587] [Q&A] [answered]
Given a 3D mesh file (.obj) with a point, make a slice across the point to intersect this 3D mesh to get a 2D slice. How to get the signed distance of this point to this 2D profile efficiently? Is there a relevant algorithm or graphics library I can refer to? Thanks a lot！

**Comment by sloriot** [comment, marked as answer]
You can get the list of intersection segments using [`Polygon_mesh_slicer`])https://doc.cgal.org/latest/Polygon_mesh_processing/classCGAL_1_1Polygon__mesh__slicer.html) and build an [`AABB_tree`](https://doc.cgal.org/latest/AABB_tree)  of segments  and the function [`squared_distance()`](https://doc.cgal.org/latest/AABB_tree/classCGAL_1_1AABB__tree.html#a24efb545081b6796d8843861a4bf49e6) to get the distance. For the sign, you need to use [`Side_of_triangle_mesh`](https://doc.cgal.org/latest/Polygon_mesh_processing/classCGAL_1_1Side__of__triangle__mesh.html). Note that you might want to compute and use only one AABB-tree for `Polygon_mesh_slicer` and `SIde_of_triangle_mesh`.

**Reply by Pupillen** [reply]
Thank you very much for your reply. If the cutting plane and the corresponding 2D slice are not fixed but change very often (1000HZ), how can I improve efficiency? I think it is relatively expensive to build the AABB tree each time. I am very demanding in terms of performance.
